pkgs/games/heroic/default.nix

```diff
@@ -1,39 +1,75 @@
-{ lib, fetchurl, appimageTools }:
-
-let
-  pname = "heroic";
-  version = "2.2.6";
-  name = "${pname}-${version}";
-  src = fetchurl {
-    url = "https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/releases/download/v${version}/Heroic-${version}.AppImage";
-    sha256 = "sha256-kL30/G4DpDPwGN7PvbWest7TcgL4Rd1c2OM4nRCT3bg=";
+{ lib
+, mkYarnPackage
+, fetchFromGitHub
+, makeWrapper
+, electron
+
+, autoPatchelfHook
+, stdenv
+, zlib
+}:
+
+mkYarnPackage rec {
+  pname = "heroic-original";
+  version = "2.4.0";
+
+  src = fetchFromGitHub {
+    owner = "Heroic-Games-Launcher";
+    repo = "HeroicGamesLauncher";
+    rev = "v${version}";
+    sha256 = "sha256-x0aJIsoKYo0aBtxglZpT5EMgLTqSekf+noEBq/0Wq2s=";
   };
-  appimageContents = appimageTools.extractType2 { inherit name src; };
 
-in
-appimageTools.wrapType2 {
-  inherit name src;
+  packageJSON = "${src}/package.json";
+  yarnLock = "${src}/yarn.lock";
```

Comment:
Thats IFD and not allowed. You would need to use the FOD variant or vendor the file.

Reply:
These files are now vendored.